dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowActionUtils.java

```diff
@@ -0,0 +1,308 @@
+     *
+     * @param actions List of action to filter
+     * @param user User to validate
+     * @param respectFrontEndRoles indicates if should respect frontend roles
+     * @param permissionable ContentType or contentlet to validate special workflow roles
+     * @return List<WorkflowAction>
+     */
+    @CloseDBIfOpened
+    List<WorkflowAction> filterActions(final List<WorkflowAction> actions,
+            final User user, final boolean respectFrontEndRoles,
+            final Permissionable permissionable) throws DotDataException {
+
+        if ((user != null) && roleAPI.doesUserHaveRole(user, cmsAdminRole)) {
+            Logger.debug(this, () -> "user:" + user.getUserId() + " has an admin role. returning all actions." );
+            return actions;
+        }
+
+        final List<WorkflowAction> permissionables = new ArrayList<>(actions);
+        if (permissionables.isEmpty()) {
+            Logger.debug(this, () -> " No actions were passed. " );
+            return permissionables;
+        }
+
+        final Iterator<WorkflowAction> actionsIterator = permissionables.iterator();
+        while (actionsIterator.hasNext()) {
+            final WorkflowAction action = actionsIterator.next();
+            boolean hasPermission = false;
+            if (null != permissionable) {
+                // Validate if the action has one of the workflow special roles
+                final boolean doesHavePermission = hasSpecialWorkflowPermission(user, respectFrontEndRoles, permissionable, action);
+                Logger.debug(this, () -> " Trying special roles for action "+action.getName()+" had permissions:" + BooleanUtils.toStringYesNo(doesHavePermission));
+                hasPermission = doesHavePermission;
+            }
+            // Validate if has other role permissions
+            if (doesUserHavePermission(action, PermissionAPI.PERMISSION_USE, user, respectFrontEndRoles)) {
+                Logger.debug(this, () -> " Trying other roles for action "+action.getName()+" had permissions: yes");
+                hasPermission = true;
+            }
+            if (!hasPermission) {
+                actionsIterator.remove();
+            }
+        }
+
+        return permissionables;
+    }
+
+
+    /**
+     * Filters the list of actions to display according to the role passed
+     *
+     * @param actions the actions to filter by role
+     * @param role the role selected
+     * @return the list of actions once the filter operation has taken place
+     */
+    @CloseDBIfOpened
+    List<WorkflowAction> filterActions(final List<WorkflowAction> actions, final Role role,
+            final Permissionable permissionable) throws DotDataException {
+
+        final List<WorkflowAction> permissionables = new ArrayList<>(actions);
+        if (permissionables.isEmpty()) {
+            Logger.debug(this, () -> " No actions were passed. " );
+            return permissionables;
+        }
+
```

Comment:
I think we are missing the validation for the `cmsAdminRole`, meaning, if the _role_ is a `CMS Admin Role` we return all the actions.

Reply:
isn't that role mapped to a user? if so then the case is handled. If not then you're right.